server/src/api/modules/snapshots/snapshot.utils.ts

```diff
@@ -204,6 +210,87 @@ function average(snapshots: Snapshot[]): Snapshot {
   return base as Snapshot;
 }
 
+/**
+ * Gets the best player/snapshot data for each metric.
+ */
+function getBestInEachMetric(snapshots: Snapshot[], players: Player[]): BestGroupSnapshot {
+  if (!snapshots || snapshots.length === 0) {
+    throw new ServerError('Invalid snapshots list. Failed to find best players.');
+  }
+
+  if (!players || players.length !== snapshots.length) {
+    throw new ServerError('Invalid players list. Failed to find best players.');
+  }
+
+  return {
+    skills: Object.fromEntries(
+      SKILLS.map(s => {
+        const valueKey = getMetricValueKey(s);
+        const snapshot = [...snapshots].sort((x, y) => y[valueKey] - x[valueKey])[0];
```

Comment:
will this .sort throw if length=1 ?

Reply:
Don't think so.
![image](https://user-images.githubusercontent.com/51417989/221363334-b08d02f0-555e-48ee-811b-7e5132b7e89a.png)